### PR TITLE
Environment.NewLine Plug and tests

### PR DESCRIPTION
**Pull request by zarlo**

### CODE REVIEW

**Comment by jp2masa** [comment]
I think the VBE changes shouldn't be applied, as @MyvarHD said in #460 and also because it's not the purpose of this Pull Request.

**Comment by charlesbetros** [comment]
I agree. Let's try to stick to one issue one pr.

**Comment by fanoI** [comment]
This other PR https://github.com/CosmosOS/Cosmos/pull/480 contains the test for this? If so could you add to this PR so AppVoyer properly check that the plug is working.

Thank you!

**Comment by fanoI** [comment]
The test is failing but I'd say that is replace_test that has not the expected value is ""That is a test string." instead should be "this a test string" (or if you prefer could change the replace test with:
test = @"That is a test strin


g";)

**Comment by jp2masa** [comment]
Try replacing `string new_expected = "this is a test string..";` with `string new_expected = "this is a test strin..g";`

**Comment by fanoI** [comment]
OK now the test is passing but when you use a real newline it not worked right?

That is in the string:
"this is a test string.
"
after the '.' there was not the same thing as Environment.NewLine? While in code is more correct how you have written if you had read it from a file you get a string similar to that probably...

File: Tests/Cosmos.Compiler.Tests.Bcl/System/StringTest.cs [hunk 1]
```diff
@@ -57,6 +57,18 @@ public static void Execute()
...
             Assert.IsTrue((test.Replace("This", "That") == replace_test), "string.Replace(string, string) does not work.");
 
+            test = "this is a test string." + Environment.NewLine;
+            string new_expected = "this is a test string..";
+            Assert.IsTrue((test.Replace(Environment.NewLine, ".") == new_expected), "string.Replace(Environment.NewLine, string) does not work.(Environment.NewLine)");
+
+			
+            test = @"this is a test string.
+			";
+            string new_expected = "this is a test string..";
```
Comment (jp2masa) [inline Tests/Cosmos.Compiler.Tests.Bcl/System/StringTest.cs]: Replace "string new_expected" with "new_expected"

Comment (zarlo) [inline Tests/Cosmos.Compiler.Tests.Bcl/System/StringTest.cs]: o how did i not see that

Comment (zarlo) [inline Tests/Cosmos.Compiler.Tests.Bcl/System/StringTest.cs]: note to self do not code in notepad

**Comment by zarlo** [comment]
it has stopped crashing YAY you would not get an error i would just crash when you used 
```
Environment.NewLine
```
plug not neded

**Pull request closed without merging**